python/setup.py

```diff
@@ -6,11 +6,11 @@
     version='0.1.0',
     author_email='ellisbigelow@google.com',
     license='../../LICENSE.txt',
-    url='https://github.com/kubeflow/kfserving/python/kfserving/kfserver',
+    url='https://github.com/kubeflow/kfserving/python/kfserving/kfserving',
     description='Model Server for arbitrary python ML frameworks.',
     long_description=open('README.md').read(),
     python_requires='>3.4',
-    packages=find_packages("kfserver"),
+    packages=find_packages("kfserving"),
```

Comment:
I agree with this naming convention :). I wonder if we can pull this out a level so that we don't have doubly nested kfserving.

I think the config file is completely fine for multiple setup calls -- at least it was working for me.

/python/setup.py
/python/kfserving/...
/python/xgbserver/...

Reply:
Is the intention to have separate pypi packages for kfserving and xgbserver as installed packages?
When I run `pip install -e .` it only says its installed xgbserver which I thought was strange though it does install both. Can you point me to the convention on this and how we would push the modules to pypi?

Reply:
kfserving and xgbserver are definitely separate packages with separate dependencies. I don't think we actually want to publish the xgbserver at all. Perhaps we can only do setup.py for kfserving.

I hadn't gotten that far ;). It's totally possible I'm doing a pypi anti-pattern, as I've never published a package before. I'll defer to your judgement.

Reply:
ok - so the xgbserver would just be added to a Dockerfile?

 I'm not sure what is the best way to handle this with the requirements for xgbserver - maybe a requirements_xgbserver.txt or something?

Reply:
I think the `setup.py` should be in each submodule and  in xgbserver's `setup.py` you can add 
```
install_requires=[
'kfserving>=0.1.0'
]
```

Reply:
So like I originally proposed ?
```
   python/kfserving/setup.py
   python/kfserving/kfservng/<python module code>
   python/xgbserver/setup.py
   python/xgbserver/xgbserver/<python module code>
```

We can still just publish to pypi only `kfserving` I suppose

Reply:
ye this looks good to me.

Reply:
SGTM. We will separate this into it's own repo soon. I need to coordinate with Abhishek. Any recommendations on name? 
kfserving-core? 
kfservers?

Reply:
i would take the same approach as kubeflow core repo, and the operators for tensorflow/pytorch etc go in their own repos. In that case, this this becomes kfserving, and rest of them can xgboost-serving etc...